File: nomad-samples/python/basic-live-channel-1.1/live_channel/start_live_channel.py
from constants.project_constants import *
from exceptions.api_exception_handler import *
from live_channel.live_channel_statuses import *
from live_channel.wait_live_channel_status import *

from libraries import json, requests
'''
 * Start a live channel
 *
 * @param {string} AUTH_TOKEN    | Authentication token
 * @param {string} CHANNEL_ID    | The ID of the channel to start
'''
async def start_live_channel(AUTH_TOKEN, CHANNEL_ID):
    # Check for valid parameters
    if (not AUTH_TOKEN or not CHANNEL_ID):
        raise Exception("Start Live Channel: Invalid API call")


    # Create header for the request
    HEADERS = {
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    try:
        # Send the request
        RESPONSE = requests.post(SERVER_URL + "/liveChannel/" + CHANNEL_ID + "/start", headers= HEADERS)

    
    
        # Wait for the Live Channel to be running
        await wait_for_live_channel_status(AUTH_TOKEN, CHANNEL_ID, LIVE_CHANNEL_STATUSES["Running"], 120, 2)
        return

    except:
        raise Exception("Start Live Channel " + CHANNEL_ID + " failed\n" + json.dumps(RESPONSE))

File: nomad-samples/admin/js/advanced-live-channel-1.1/live-channel/start-live-channel.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";
import liveChannelStatuses from "./live-channel-statuses.js";
import waitForLiveChannelStatus from "./wait-live-channel-status.js";

/**
 * Start a live channel
 *
 * @param {string} authToken    | Authorization token
 * @param {string} channelId    | The ID of the channel to start
 */
export default async function startLiveChannel(authToken, channelId) {
    // Check for valid parameters
    if (!authToken || !channelId) {
        throw new Error("Start Live Channel: Invalid API call");
    }

    // Create header for the request
    const headers = new Headers();
    headers.append("Authorization", `Bearer ${authToken}`);

    // Send the request
    const response = await fetch(`${prjConstants.SERVER_URL}/liveChannel/${channelId}/start`, {
        method: "POST",
        headers: headers
    });

    // Check for success
    if (response && response.ok) {
        // Wait for the Live Channel to be running
        await waitForLiveChannelStatus(authToken, channelId, liveChannelStatuses.Running, 120, 2);
        return;
    }

    await apiExceptionHandler(response, `Start Live Channel ${channelId} failed`);
}
